build(docs-infra): update docs examples to Angular v11.2.10

petebacondarwin opened the pull request:
This commit updates the docs examples to Angular v11.2.10.
See the [diff][1] between 11.0.1 and 11.2.9 (CLI) and 11.2.10 (FW).

[1]: https://github.com/cexbrayat/angular-cli-diff/compare/11.2.9..11.0.1

petebacondarwin committed:
build(docs-infra): update docs examples to Angular v11.2.10

This commit updates the docs examples to Angular v11.2.10. See the [diff between 11.0.1 and 11.2.10 (FW) and 11.2.9 (CLI)][1].

The changes are fairly trivial including:

- Removal of `emitDecoratorMetadata` from tsconfig.json files, where no JIT compilation is required.
- Setting `enableI18nLegacyMessageIdFormat` to `false` for CLI based applications - the i18n example was already migrated away from legacy message IDs.

[1]: https://github.com/cexbrayat/angular-cli-diff/compare/11.2.9..11.0.1

gkalpak reviewed (approved):
The diff URL is a little misleading since there is no `11.2.10` version for CLI yet. There _are_ some changes between CLI versions [11.0.2 and 11.2.9][1], which might require changes to the boilerplate files. Other than that, lgtm (as soon as CI is green) ✔️ 

[1]: https://github.com/cexbrayat/angular-cli-diff/compare/11.0.2..11.2.9

petebacondarwin commented:
> The diff URL is a little misleading since there is no `11.2.10` version for CLI yet. There _are_ some changes between CLI versions [11.0.2 and 11.2.9](https://github.com/cexbrayat/angular-cli-diff/compare/11.0.2..11.2.9), which might require changes to the boilerplate files. Other than that, lgtm (as soon as CI is green) ✔️

Oops, yes I actually meant for the link to be for 11.2.9 (there is a mismatch between the latest versions of framework - 11.2.10 - and CLI - 11.2.9). I did check this diff manually - the changes in that corrected diff are most only formatting and an extra comment added to the karma config. 

One change that I didn't spot until now is that the projects will default to `enableI18nLegacyMessageIdFormat: false`. I'll update that now and see if it breaks any of the i18n examples...

petebacondarwin committed:
docs: migrate example to use non-legacy message IDs

From 11.x, new projects are set up to use the latest message ID format.
This commit updates the i18n example project to stop using legacy IDs.

petebacondarwin committed:
refactor(docs-infra): tidy up systemjs boilerplate for upgrade examples

The ngUpgrade examples mostly rely upon SystemJS configuration.
This commit tidies up how these examples are built and tested so that
it will be easier to migrate them to work with Angular 11.2.x

gkalpak commented on aio/tools/examples/shared/boilerplate/systemjs/package.json:
```diff
@@ -4,16 +4,14 @@
     "serve:e2e": "lite-server -c=bs-config.e2e.json",
     "prestart": "npm run build",
     "start": "concurrently \"npm run build:watch\" \"npm run serve\"",
-    "pree2e": "webdriver-manager update && npm run build:e2e",
```
Why remove this?

petebacondarwin replied:
The TS spec can be consumed directly by Protractor because it has the `ts-node` plugin loaded.
As far as the `webdriver-manager` goes, I always find this step really annoying; it is normally dealt with when the packages are installed, no?

gkalpak replied:
I don't think it is taken care off automatically. We could add it as a `postinstall` script to avoid having it run on every `yarn e2e` invocation.
(There is also a (tiny 😛) I am wring and it is being taken care of automatically.)

petebacondarwin replied:
We have this:

https://github.com/angular/angular/blob/4a33879ebd4fe761041b4a942cb33f2c2b9efcac/aio/tools/examples/shared/package.json#L9-L10

gkalpak replied:
But that is only useful on CI. Omitting the script from the actual `package.json` in the example, makes it not self-contained.
(It doesn't make any real difference atm, but I still think it is a good idea to include it :grin: - I have created #41796.)

gkalpak commented on aio/content/examples/upgrade-phonecat-1-typescript/tsconfig.json:
```diff
@@ -1,26 +0,0 @@
-  "exclude": [
-    "node_modules/*",
-    "**/*-aot.ts",
-    "aot/**/*"
```
Is this line not needed any more?

petebacondarwin replied:
This now comes from the systemjs boilerplate.

gkalpak replied:
The systemjs boilerplate does not contain a exclusion for `aot/**`: https://github.com/angular/angular/blob/de20566aa6c81bc164bd76365173ab33bbf325cd/aio/tools/examples/shared/boilerplate/systemjs/tsconfig.json#L16-L20

(But it could be the case that it is not needed any more...)

gkalpak replied:
But files are copied into that directory by [copy-dist-files.js][1] :grin:
I think it would be better to keep excluding it (even if just to err on the side of caution) :wink: (I have created #41796.)

[1]: https://github.com/angular/angular/blob/73105aa7d6cf1116a6a9f103bd074f65db7584fe/aio/content/examples/upgrade-phonecat-2-hybrid/copy-dist-files.js#L24-L25

petebacondarwin replied:
But the values that are copied in are not `.ts` files so they don't get compiled anyway, since we haven't got `allowJs` set to true.

petebacondarwin replied:
But I guess it won't do any harm to add this back for all these system examples.

gkalpak replied:
> But the values that are copied in are not `.ts` files

Ah, true! Somehow I thought we were copying the `app/` directory (which contains `.ts` files), but now I see we only copy specific sub-directories (`app/img/`, `app/phones/`) that do not contain `.ts` files.

I'll remove it from my PR then (it doesn't hurt, but it is not needed either, so... 😄)

gkalpak commented on aio/tools/examples/run-example-e2e.js:
```diff
@@ -194,13 +192,7 @@ function runProtractorSystemJS(prepPromise, appDir, appRunSpawnInfo, outputFile)
-              '--params.appDir=' + appDir, '--params.outputFile=' + outputFile
-            ],
-            {cwd: SHARED_PATH});
+        const spawnInfo = spawnExt('yarn', [ 'protractor' ], {cwd: appDir});
```
I would keep the `outputFile` param, since `run-examples-e2e` uses it to log stuff. It currently points to the same file as `defaultOutputFile` in `boilerplate/systemjs/protractor.config.js`, but this could easily change in the future without us noticing:

```suggestion
        const spawnInfo = spawnExt('yarn', [ 'protractor', '--params.outputFile=' + outputFile ], {cwd: appDir});
```

petebacondarwin replied:
If we always set the value (and to the same value as the default), then I would recommend removing the default, no?

gkalpak replied:
Keeping the default value (or skipping logging when there is no `outputFile` specified) would be useful for when we want to test an example locally. I think it is useful to be able to just `cd` into an example directory and run `yarn e2e` (or `yarn protractor`).

gkalpak reviewed (approved):
Nice :tada:

petebacondarwin committed:
fixup! refactor(docs-infra): tidy up systemjs boilerplate for upgrade examples

gkalpak reviewed (approved):
Reviewed-for: global-docs-approvers

AndrewKushnir commented:
@petebacondarwin FYI there was a conflict merging this PR into `11.2.x` (patch) branch, so I've merged this change to master and rc (`12.0.x`) branches. If thus change needs to go into the `11.2.x` (patch) branch, please create a new PR with the necessary changes. Thank you.
